src/sagemaker/jumpstart/accessors.py

```diff
@@ -270,10 +278,34 @@ def get_model_specs(
         if s3_client is not None:
             additional_kwargs.update({"s3_client": s3_client})
 
+        if hub_arn:
+            additional_kwargs.update({"sagemaker_session": sagemaker_session})
+
         cache_kwargs = JumpStartModelsAccessor._validate_and_mutate_region_cache_kwargs(
             {**JumpStartModelsAccessor._cache_kwargs, **additional_kwargs}
         )
         JumpStartModelsAccessor._set_cache_and_region(region, cache_kwargs)
+
+        if hub_arn:
+            try:
+                hub_model_arn = construct_hub_model_arn_from_inputs(
+                    hub_arn=hub_arn, model_name=model_id, version=version
+                )
+                model_specs = JumpStartModelsAccessor._cache.get_hub_model(
+                    hub_model_arn=hub_model_arn
+                )
+                model_specs.set_hub_content_type(HubContentType.MODEL)
+                return model_specs
+            except:  # noqa: E722
```

Comment:
nit: We can catch the specific exception here.

Reply:
nit: We can check model-ref first, assuming there are more chances Hub has more refs than models.